Source/Task/WaitTimer_stl.cpp

```diff
@@ -132,7 +145,7 @@ void TimerQueue::Worker() noexcept
             TimerEntry entry = Pop();
 
             // release the lock while invoking the callback, just in case timer
-            // gets destroyed on this thread or readds itself in the callback
+            // gets destroyed on this thread or reads itself in the callback
```

Comment:
readds (adds again) was correct, we are worried about the callback interacting with TimerQueue

Reply:
Works due to lock